Internal dispatcher to protect against starvation (#26816)
akka/akka: akka-docs/src/main/paradox/project/migration-guide-2.5.x-2.6.x.md

Actor DSL is a rarely used feature and has been deprecated since `2.5.0`.
Use plain `system.actorOf` instead of the DSL to create Actors if you have been using it.

### Internal dispatcher introduced

To protect the Akka internals against starvation when user code blocks the default dispatcher (for example by accidental

> use of blocking APIs from actors) a new internal dispatcher has been added. All internal actors that Akka uses that are non blocking

Reviewer: ```suggestion
use of blocking APIs from actors) a new internal dispatcher has been added. All of Akka's internal, non-blocking actors now run on the internal dispatcher by default.
```

> are now by default run on this separate dispatcher.

Reviewer: ```suggestion
```

The dispatcher can be configured through `akka.actor.internal-dispatcher`.

> For systems where you want to keep running the internal actors on a single dispatcher, for example to keep the resource

Reviewer: Here a slight rephrasing that might be more explicit about why you would want to change back to the old behavior and its risks:

> For maximum performance, you might want to use a single shared dispatcher for all non-blocking, asynchronous actors, user actors and Akka internal actors. In that case, can configure the `akka.actor.internal-dispatcher` with a string value of `akka.actor.default-dispatcher`. This reinstantiates the behavior from previous Akka versions but also removes the isolation between user and Akka internals. So, use at your own risk!
